### PR TITLE
Support chmod for jni-fuse

### PR DESCRIPTION
**Pull request by maobaolong**
Now jni-based fuse lack chmod operation implementation, this PR aim to complete it.

### CODE REVIEW

**Comment by maobaolong** [comment]
@apc999 @LuQQiu Could you please take a look at this PR? Thanks

File: integration/fuse/src/main/native/libjnifuse/jnifuse_fs.h [hunk 1]
```diff
@@ -31,6 +31,7 @@ class RmdirOperation;
 class UnlinkOperation;
 class WriteOperation;
 class RenameOperation;
+class ChmodOperation;
```
Comment (apc999) [inline integration/fuse/src/main/native/libjnifuse/jnifuse_fs.h:34]: please keep these alphabetical

File: integration/fuse/src/main/native/libjnifuse/jnifuse_fs.h [hunk 2]
```diff
@@ -62,6 +63,7 @@ class JniFuseFileSystem {
   RmdirOperation *rmdirOper;
   WriteOperation *writeOper;
   RenameOperation *renameOper;
+  ChmodOperation *chmodOper;
```
Comment (apc999) [inline integration/fuse/src/main/native/libjnifuse/jnifuse_fs.h:66]: same here

File: integration/fuse/src/main/native/libjnifuse/jnifuse_helper.cc [hunk 3]
```diff
@@ -58,6 +58,7 @@ JNIEXPORT jint JNICALL Java_alluxio_jnifuse_LibFuse_fuse_1main_1real(
   jnifuse_oper.rmdir = rmdir_wrapper;
   jnifuse_oper.write = write_wrapper;
   jnifuse_oper.rename = rename_wrapper;
+  jnifuse_oper.chmod = chmod_wrapper;
```
Comment (apc999) [inline integration/fuse/src/main/native/libjnifuse/jnifuse_helper.cc:61]: ditto

File: integration/fuse/src/main/native/libjnifuse/jnifuse_impls.cc [hunk 4]
```diff
@@ -89,3 +89,7 @@ int write_wrapper(const char *path, const char *buf, size_t size, off_t off,
 int rename_wrapper(const char *oldPath, const char *newPath) {
   return jnifuse::JniFuseFileSystem::getInstance()->renameOper->call(oldPath, newPath);
 }
+
+int chmod_wrapper(const char *path, mode_t mode) {
```
Comment (apc999) [inline integration/fuse/src/main/native/libjnifuse/jnifuse_impls.cc:93]: ditto

File: integration/fuse/src/main/native/libjnifuse/jnifuse_impls.h [hunk 5]
```diff
@@ -32,5 +32,6 @@ int rmdir_wrapper(const char *path);
 int write_wrapper(const char *path, const char *buf, size_t size, off_t off,
                   struct fuse_file_info *fi);
 int rename_wrapper(const char *oldPath, const char *newPath);
+int chmod_wrapper(const char *path, mode_t mode);
```
Comment (apc999) [inline integration/fuse/src/main/native/libjnifuse/jnifuse_impls.h:35]: ditto

**Review by apc999 — APPROVED** [review]
Only comment is to keep the new definitions alphabetical in their source code

**Comment by maobaolong** [comment]
@apc999 Thanks for your review, I fix all the related alphabetical issues. PTAL.

File: integration/fuse/src/main/native/libjnifuse/jnifuse_impls.h (hunk 5 above)
Comment (maobaolong) [reply integration/fuse/src/main/native/libjnifuse/jnifuse_impls.h]: done.

File: integration/fuse/src/main/native/libjnifuse/jnifuse_impls.cc (hunk 4 above)
Comment (maobaolong) [reply integration/fuse/src/main/native/libjnifuse/jnifuse_impls.cc]: done.

File: integration/fuse/src/main/native/libjnifuse/jnifuse_helper.cc (hunk 3 above)
Comment (maobaolong) [reply integration/fuse/src/main/native/libjnifuse/jnifuse_helper.cc]: done.

File: integration/fuse/src/main/native/libjnifuse/jnifuse_fs.h (hunk 2 above)
Comment (maobaolong) [reply integration/fuse/src/main/native/libjnifuse/jnifuse_fs.h]: done.

File: integration/fuse/src/main/native/libjnifuse/jnifuse_fs.h (hunk 1 above)
Comment (maobaolong) [reply integration/fuse/src/main/native/libjnifuse/jnifuse_fs.h]: done.

**Review by apc999 — APPROVED** [review]
LGTM. Thanks for the fix!

**Comment by apc999** [comment]
failed test is unrelated

**Comment by apc999** [comment]
alluxio-bot, merge this please

**Pull request merged**